output/transform/postgres_plans.go

```diff
@@ -0,0 +1,164 @@
+package transform
+
+import (
+	"time"
+
+	snapshot "github.com/pganalyze/collector/output/pganalyze_collector"
+	"github.com/pganalyze/collector/state"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type planKey struct {
+	queryIdx int32
+	planID   int64
+}
+
+type planValue struct {
+	plan      state.PostgresPlan
+	planStats state.DiffedPostgresStatementStats
+}
+
+func groupPlans(plans state.PostgresPlanMap, statsMap state.DiffedPostgresPlanStatsMap, queryIDKeyToIdx QueryIDKeyToIdx) map[planKey]planValue {
+	// We group multiple different queryIDs with the same fingerprints with one "statementKey"
+	// (e.g. SELECT * FROM users WHERE name in ($1) and SELECT * FROM users WHERE name in ($1, $2) have the same fingerprints even though queryID will be different)
+	// With query plans, we also group plans and plan stats with a statementKey + planID.
```

Comment:
Example (different query ID, same fingerprint, same plan ID - we group these two plans together):

```
-[ RECORD 1 ]-----------------------------------------------------------------------------
queryid | 1649607691878460349
planid  | 683693185
query   | select queryid, planid from aurora_stat_plans() where query in ($1) limit $2
-[ RECORD 2 ]-----------------------------------------------------------------------------
queryid | 8308867196947722868
planid  | 683693185
query   | select queryid, planid from aurora_stat_plans() where query in ($1, $2) limit $3
```